### PR TITLE
Ignore .condarc files during installation

### LINKED ISSUE #542: Bundled `conda-standalone` will inherit settings from existing .condarc files
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

Our bundled conda-standalone.exe will read and load existing `.condarc` configurations, which might break the solver behavior and possibly others. For example, we built the installer with specs from different channels with `channel_priority: flexible` (default) but the user running the installer has a .condarc file with `channel_priority: strict` -- potential problems.

The solution is to add a flag or environment variable to upstream `conda` to disable the search path for `.condarc` files.

### Conda Info

```shell
N/A
```

### Conda Config

```shell
N/A
```

### Conda list

```shell
N/A
```

### Additional Context

This could go to `conda/conda-standalone` too, but I'll leave it here as a reminder to enable once available, since we are the primary users of that tool.

### PR DESCRIPTION
**Pull request by marcoesters**
### Description

`conda-standalone` and `micromamba` load existing `.condarc` files during the installation process, which can have unintended consequences and even break the installation. These issues have been frequently reported (see #542, #568, and the issues reported therein). The behavior of the installer should be set by the installer itself and not by residual configuration files, especially since they are not cleaned up at all (see #642).

`micromamba` has a `--no-rc` option, which is also proposed to be added to `conda-standalone` (https://github.com/conda/conda-standalone/pull/99). This feature will add the `--no-rc` option with compatible standalone executables.

Closes #542 and #568.

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](https://github.com/conda/constructor/blob/main/news/TEMPLATE)) for the next release's release notes?
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

### CODE REVIEW

File: constructor/main.py [hunk 1]
```diff
@@ -197,6 +197,15 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
...
 
+    # Add --no-rc option to CONDA_EXE command so that existing
+    # .condarc files do not pollute the installation process.
+    if exe_name == "conda-standalone" and Version(exe_version) >= Version("24.9.0"):
+        info["_ignore_condarcs_arg"] = "--no-rc"
+    elif exe_name == "micromamba":
+        info["_ignore_condarcs_arg"] = "--no-rc"
+    else:
+        info["_ignore_condarcs_arg"] = ""
+
```
Comment (jezdez) [inline constructor/main.py:208]: That hardcoding of the names of binaries that constructor builds is a dealbreaker, let's find a different way

**Review by jezdez — CHANGES_REQUESTED** [review]

File: constructor/main.py (hunk 1 above)
Comment (marcoesters) [reply constructor/main.py:208]: Is your objection here that I use an if-else chain or that I use strings? The latter is the result of the `identify_conda_exe` function, so changing that will widen the scope of this PR.

Identifying whether the standalone function is `conda-standalone` is done inconsistently (see, e.g., here: https://github.com/conda/constructor/blob/49b80727bbdabce4d150e3fba4d03c953574ea1f/constructor/main.py#L95), but I wonder if a separate PR would be a better place to address the issue.

Comment (marcoesters) [reply constructor/main.py:208]: Here is a PR that addresses the issue: https://github.com/conda/constructor/pull/864

That would get rid of string comparisons at least and removes inconsistencies.

File: constructor/main.py [hunk 2]
```diff
@@ -197,6 +197,15 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
             )
         )
 
+    # Add --no-rc option to CONDA_EXE command so that existing
+    # .condarc files do not pollute the installation process.
+    if exe_name == "conda-standalone" and Version(exe_version) >= Version("24.9.0"):
```
Comment (jaimergp) [inline constructor/main.py:202]:
```suggestion
    if exe_name == "conda-standalone" and exe_version and Version(exe_version) >= Version("24.9.0"):
```

Protect against `exe_version = None`.

**Comment by marcoesters** [comment]
Even if approved, we should probably wait with merging this until `conda-standalone 24.9.x` is released.

**Review by jaimergp — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**